Sources/Filters/General/ImageDataOutlineFilter/index.js

```diff
@@ -0,0 +1,149 @@
+import { vec3 } from 'gl-matrix';
+
+import macro from 'vtk.js/Sources/macros';
+import vtkCubeSource from 'vtk.js/Sources/Filters/Sources/CubeSource';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkCellArray from '../../../Common/Core/CellArray';
+
+const { vtkErrorMacro } = macro;
+
+// prettier-ignore
+export const LINE_ARRAY = [
+  2, 0, 1,
+  2, 2, 3,
+  2, 4, 5,
+  2, 6, 7,
+  2, 0, 2,
+  2, 1, 3,
+  2, 4, 6,
+  2, 5, 7,
+  2, 0, 4,
+  2, 1, 5,
+  2, 2, 6,
+  2, 3, 7,
+];
+
+// ----------------------------------------------------------------------------
+// vtkImageDataOutlineFilter methods
+// ----------------------------------------------------------------------------
+
+function vtkImageDataOutlineFilter(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkImageDataOutlineFilter');
+
+  function transformPoints(points, transform) {
+    const tmp = [0, 0, 0];
+    for (let i = 0; i < points.length; i += 3) {
+      const p = points.subarray(i, i + 3);
+      if (transform.length === 9) {
+        vec3.transformMat3(tmp, p, transform);
+      } else {
+        vec3.transformMat4(tmp, p, transform);
+      }
+      [p[0], p[1], p[2]] = tmp;
+    }
+  }
+
+  publicAPI.requestData = (inData, outData) => {
+    // implement requestData
+    const input = inData[0];
+
+    if (!input || !input.isA('vtkImageData')) {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+
+    if (!model.tmpOut) {
+      // allocate output object if not done previously.
+      model.tmpOut = vtkPolyData.newInstance();
+    }
+
+    // First create a cube polydata in the index-space of the image.
+    const spatialExt = input.getSpatialExtent();
+    if (!spatialExt) {
+      vtkErrorMacro('Unable to fetch spatial extents of input image.');
+      return;
+    }
+
+    model.cubeSource.setXLength(spatialExt[1] - spatialExt[0]);
+    model.cubeSource.setYLength(spatialExt[3] - spatialExt[2]);
+    model.cubeSource.setZLength(spatialExt[5] - spatialExt[4]);
+
+    model.cubeSource.setCenter(
+      model.cubeSource.getXLength() / 2.0,
+      model.cubeSource.getYLength() / 2.0,
+      model.cubeSource.getZLength() / 2.0
+    );
```

Comment:
(optional) you may want to use `vtkBoundingBox` to avoid having to compute those lengths and center manually.

Reply:
Updated to use `vtkBoundingBox .getLengths()` and `.getCenter()`